jbt-it/neomdb
# 301 upgrade to react router v6

Upgraded react router v5 to v6. Created three new Route components based on the outdated methods from v5. Created a new Layout component that renders the menu drawer on the left and the content of a page on the right side. Therefore removed the offset from the css class content-page. And created a new auth hook.


How to test:
- Check if the private routes are reachable without being logged in => expected outcome: private routes are not reachable if not logged in
- Check if the public routes such as /login are reachable  => expected outcome: public routes should only be reachable if not logged in
- Check if the protected route "/berechtigungen" is reachable when having permissions and if it is not reachable with a user without permissions
- Generally check if the content of the pages is correctly shown
